packages/cli/src/commands/hydrogen/deploy.ts

```diff
@@ -303,12 +301,22 @@ export async function oxygenDeploy(
     );
   }
 
+  let fallbackEnvironmentTag = branch;
+
+  // If the user has explicitly selected `Preview` then we should not pass an
+  // environment tag at all.
+  if (deploymentEnvironmentTag === 'shopify-preview-environment') {
+    fallbackEnvironmentTag = undefined;
+    deploymentEnvironmentTag = undefined;
+  }
+
   const config: DeploymentConfig = {
     assetsDir: 'dist/client',
     bugsnag: true,
     deploymentUrl,
     deploymentToken: parseToken(token as string),
-    environmentTag: environmentTag || deploymentEnvironmentTag || branch,
+    environmentTag:
+      environmentTag || deploymentEnvironmentTag || fallbackEnvironmentTag,
```

Comment:
I haven't actually tested to double check but I think oxygen cli will override this still and set it to the branch from the environment [here](https://github.com/Shopify/oxygen-cli/blob/main/src/deploy/index.ts#L62)

Reply:
If we eliminate all places that we send _something_ though (or send some garbage that could never be a branch 🤔 ) [this](https://github.com/Shopify/oxygen-hub/blob/7ddb881a27c9239c8e775019e3307c1f45911a8a/app/graphql/graph_api/v2/mutations/deployment_initiate.rb#L108C27-L110) is where in oxygen hub we load the default environment

Reply:
One other possible thing we could do is have an explicit `previewEnvironment`/`defaultEnvironment` option maybe? [here](https://github.com/Shopify/oxygen-cli/blob/main/src/deploy/types.ts#L28-L49) that would make sure oxygen-cli never sends a tag.